### ISSUE TITLE
String concatenation fo literals and columns causes problems

### ISSUE THREAD

**Issue by laakejs** [pola-rs/polars#944]
#### Are you using Python or Rust?

**Python**

#### What version of polars are you using?

0.8.9

#### What operating system are you using polars on?

Python 3.6.9 on Ubuntu 18.04 (WSL)

#### Describe your bug.

When concatenating literal and column Expressions to a single string only the first entry of the column Expression is used.

#### What are the steps to reproduce the behavior?

```python
import polars as pl
df = pl.DataFrame(
    {
        "nrs": [1, 2, 3, 4],
        "name": ["ham", "spam","foo", None],
    }
)
print(df.with_column((pl.lit("Dr. ") + pl.col("name")).alias("graduated name")))
```

#### What is the actual behavior?

```
shape: (4, 3)
╭─────┬────────┬────────────────╮
│ nrs ┆ name   ┆ graduated name │
│ --- ┆ ---    ┆ ---            │
│ i64 ┆ str    ┆ str            │
╞═════╪════════╪════════════════╡
│ 1   ┆ "ham"  ┆ "Dr. ham"      │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2   ┆ "spam" ┆ "Dr. ham"      │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3   ┆ "foo"  ┆ "Dr. ham"      │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 4   ┆ null   ┆ "Dr. ham"      │
╰─────┴────────┴────────────────╯

```

#### What is the expected behavior?

Use the individual Row.

```
shape: (4, 3)
╭─────┬────────┬────────────────╮
│ nrs ┆ name   ┆ graduated name │
│ --- ┆ ---    ┆ ---            │
│ i64 ┆ str    ┆ str            │
╞═════╪════════╪════════════════╡
│ 1   ┆ "ham"  ┆ "Dr. ham"      │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2   ┆ "spam" ┆ "Dr. spam"     │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3   ┆ "foo"  ┆ "Dr. foo"      │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 4   ┆ null   ┆ "Dr."          │
╰─────┴────────┴────────────────╯
```
or (I am not sure about null handling here)
```
shape: (4, 3)
╭─────┬────────┬────────────────╮
│ nrs ┆ name   ┆ graduated name │
│ --- ┆ ---    ┆ ---            │
│ i64 ┆ str    ┆ str            │
╞═════╪════════╪════════════════╡
│ 1   ┆ "ham"  ┆ "Dr. ham"      │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2   ┆ "spam" ┆ "Dr. spam"     │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3   ┆ "foo"  ┆ "Dr. foo"      │
├╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 4   ┆ null   ┆ null           │
╰─────┴────────┴────────────────╯
```

**Comment by ritchie46** [comment]
Thanks for the issue report. Fixed in #945.

**Comment by ritchie46** [comment]
Fixed and released :+1:
